homeassistant/components/flo/device.py

```diff
@@ -58,7 +58,9 @@ def id(self) -> str:
     @property
     def device_name(self) -> str:
         """Return device name."""
-        return f"{self.manufacturer} {self.model}"
+        return self._device_information.get(
+            "nickname", f"{self.manufacturer} {self.model}"
+        )
```

Comment:
Usually you'll have multiple leak detectors, so using the device nickname is important to get sensible device names.